src/Microsoft.ML.Dnn/DnnRetrainTransform.cs

```diff
@@ -379,14 +379,18 @@ private void TrainCore(DnnRetrainEstimator.Options options, IDataView input, IDa
             ITensorValueGetter[] srcTensorGetters,
             Runner runner)
         {
-            float loss = 0;
-            float metric = 0;
+            float loss = 0.0f;
+            float metric = 0.0f;
             for (int i = 0; i < inputs.Length; i++)
                 runner.AddInput(inputs[i], srcTensorGetters[i].GetBufferedBatchTensor());
 
             Tensor[] tensor = runner.Run();
-            loss = tensor.Length > 0 && tensor[0] != IntPtr.Zero ? (float)tensor[0].ToArray<float>()[0] : 0.0f;
-            metric = tensor.Length > 1 && tensor[1] != IntPtr.Zero ? (float)tensor[1].ToArray<float>()[0] : 0.0f;
+            if (tensor.Length > 0 && tensor[0] != IntPtr.Zero)
+                tensor[0].ToScalar<float>(ref loss);
+
+            if (tensor.Length > 1 && tensor[1] != IntPtr.Zero)
+                tensor[1].ToScalar<float>(ref metric);
```

Comment:
In this case it is guaranteed that you will have two tensors, see line 354-357 please remove the if condition check:
tensor[0].ToScalar<float>(ref loss);
tensor[1].ToScalar<float>(ref metric);